docs/INSTALL-devel.rst

```diff
@@ -0,0 +1,292 @@
+
+Create a new user on your system::
+  
+  useradd -m -d /home/haas_dev -s /bin/bash -c "for HaaS development" haas_dev
+
+This will create a `haas_dev` user with following attributes::
+  
+  haas_dev:x:1002:1002:for HaaS development:/home/haas_dev:/bin/bash
+
+exact uid/gid may vary depending on your system. 
+
+Switch to the `haas_dev` user::
+
+  sudo su - haas_dev
+
+Setup database and role to control it.
+=============================================
+
+Create database named `haas_dev` owned by user also named as `haas_dev`.
+
+1. Create a database role named `haas_dev` with priviledges to::
+ 
+   -r create roles
+   -d create databases and
+   -P will prompt for the password of the new user.
+   
+This is necessary since we have configured postgreSQL to use password authentication::
+
+   sudo -i -u postgres
+   -bash-4.2$ createuser -r -d -P haas_dev
+   Enter password for new role:  <Input password for database role haas>
+   Enter it again: <Retype password for role haas>
+
+
+Confirm that the role with requisite privileges is created **as postgres user**::
+
+  -bash-4.2$ psql -c '\dg'
+                             List of roles
+   Role name |                   Attributes                   | Member of
+  -----------+------------------------------------------------+-----------
+   haas_dev      | Create role, Create DB                         | {}
+   postgres  | Superuser, Create role, Create DB, Replication | {}
+
+
+If you wish to delete the user. do the following **as postgres user**::
+
+  dropuser haas
+
+**Note**: Make sure that the database role you create corresponds to an existing system user.
+eg. There has to be a system user `haas` to access database named `haas` as database role named `haas`.
+
+
+Create database `haas_dev` owned by database role `haas_dev`::
+
+  sudo -i -u haas_dev
+  -bash-4.2$ createdb haas
+
+Confirm it created a database named `haas_dev` and it is owned by `haas_dev`::
+
+
+  psql -c '\l'
+                                  List of databases
+    Name    |  Owner   | Encoding |   Collate   |    Ctype    |   Access privileges
+  -----------+----------+----------+-------------+-------------+-----------------------
+   haas_dev  | haas_dev | UTF8     | en_US.UTF-8 | en_US.UTF-8 |
+   postgres  | postgres | UTF8     | en_US.UTF-8 | en_US.UTF-8 |
+   template0 | postgres | UTF8     | en_US.UTF-8 | en_US.UTF-8 | =c/postgres          +
+             |          |          |             |             | postgres=CTc/postgres
+   template1 | postgres | UTF8     | en_US.UTF-8 | en_US.UTF-8 | =c/postgres          +
+             |          |          |             |             | postgres=CTc/postgres
+
+
+Making HAAS aware of this setup
+===============================
+
+If you have followed all steps so far.
+put following string in **haas.cfg** under section **[database]**::
+
+  uri = postgresql://haas_dev:<clear text password >@localhost:5432/haas_dev
```

Comment:
The haas.cfg is detailed much later in the guide, asking the user to add something to it now is confusing.